conjure-core/src/main/java/com/palantir/conjure/defs/validator/EndpointDefinitionValidator.java

```diff
@@ -100,17 +101,32 @@ public void validate(EndpointDefinition definition, DealiasingTypeVisitor dealia
                     .stream()
                     .filter(arg -> !arg.getParamType().accept(ParameterTypeVisitor.IS_BODY))
                     .forEach(arg -> {
-                        boolean isValid = dealiasingTypeVisitor.dealias(arg.getType())
-                                .fold(
-                                        typeDefinition -> true,
-                                        type -> !type.accept(TypeVisitor.IS_BINARY)
-                                                && !type.accept(TypeVisitor.IS_ANY)
-                                );
                         Preconditions.checkArgument(
-                                isValid, "Non body parameters cannot be of the 'binary' type: '%s' is not allowed",
+                                validateType(arg.getType(), dealiasingTypeVisitor),
+                                "Non body parameters cannot be of the 'binary' type: '%s' is not allowed",
```

Comment:
"cannot reference the 'binary' type" might be a bit more accurate, since `list<binary>` is technically not the binary type.

Reply:
Would it also be possible to print the actual type definition in the error message (`list<binary>` etc.)? If so, I think that would make the error message a bit more user-friendly.

Reply:
Good call. Rendering the human-readable type definition isn't trivial, but I've added the endpoint name and path for easier debugging.